File: LeetCode/problems/19-remove-nth-node-from-end-of-list/19-remove-nth-node-from-end-of-list.js
/**
 * Definition for singly-linked list.
 * function ListNode(val, next) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.next = (next===undefined ? null : next)
 * }
 */
/**
 * @param {ListNode} head
 * @param {number} n
 * @return {ListNode}
 */
var removeNthFromEnd = function(head, n) {
    let dummy = new ListNode(null,head);
    let pre = dummy;
    let cur = head;

    let I = 0;
    while(I<n-1){
        cur = cur.next;
        I++;
    }

    while(cur){
        if(cur.next == null){
            pre.next = pre.next.next;
            break;
        }
        
        cur = cur.next;
        pre = pre.next; 
    }

    return dummy.next;
};

File: LeetCode/problems/19-remove-nth-node-from-end-of-list/19-remove-nth-node-from-end-of-list.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
class Solution {
    public ListNode removeNthFromEnd(ListNode head, int n) {
        ListNode cur = head.next;
        ListNode pre = head;

        int I = 0;
        while(I < n && cur != null){
            cur = cur.next;
        }

        pre.next = cur;

        return head;
    }
}